src/roman_datamodels/datamodels/_core.py

```diff
@@ -56,7 +56,22 @@ def __init_subclass__(cls, **kwargs):
         # Add to registry
         MODEL_REGISTRY[cls._node_type] = cls
 
+    def __new__(cls, init=None, **kwargs):
+        """
+        Handle the case where one passes in an already instantiated version
+        of the model. In this case the constructor should just directly return
+        the model.
+        """
+        if init.__class__.__name__ == cls.__name__:
```

Comment:
Bad things happen (infinite recursion) if one tries to `isinstance` checks here.